lib/ansible/modules/clustering/kubevirt/kubevirt_vm.py

```diff
@@ -174,11 +174,16 @@
 
 RETURN = '''
 vm:
-    description:
-      - The virtual machine that user manages.
-      - It contains all attributes https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachine
-    returned: success
-    type: complex
+  description:
+    - The virtual machine managed by the user.
+  returned: success
+  type: complex
+  contains:
+    k8s_objects:
+      description:
+      - A list of kubernetes objects this module created, modified or read during operation.
+      returned: success
+      type: list
```

Comment:
There is one object in the list. How can we get more than one object?

Reply:
Yes. I didn't have time to get it completely finished yesterday, but the idea is this:
1. Once I finish the patch, it'll return a VM and potentially a VMI when operating on one.
2. If we ever add, say, built–in port mapping handling, then it also might return a Service object.
3. We've discussed supporting kubevirt longterm without having to change the module too much. This approach helps accomplish such a task, since if we ever need to support any new resource types, they just get returned as part of `k8s_objects`.

Reply:
> 1. Once I finish the patch, it'll return a VM and potentially a VMI when operating on one.

The question is why do we want to expose both so far we did good job in hiding `VMI` implementation details. What is the value for a user?

Reply:
Long term I'm afraid that if we commit to pretending only VMs exist despite that not being the case, we might end up regretting it. Full reasoning is in the other thread.